### DISCUSSION TITLE
Safety Features in Customer App

### DISCUSSION THREAD

**Discussion by Pranav161996** [nammayatri/nammayatri#339] [Ideas]
<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-d59f4413-7fff-87fb-9bd4-6c9e14548a53"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:#ffff00;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Overview:</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:#ffff00;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">As we are growing and acquiring more customers, it is being difficult for our current on-call customer support to handle all complaints and requests. We also need a proper escalation matrix along with action items on how we will resolve these issues so opening up this discussion for inputs with regards to safety features in the customer app:</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:#ffff00;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Types of Issues:</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">We have identified the following scenarios where a customer may feel unsafe/uncomfortable and may require our assistance</span></p><ul style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:disc;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Rash Driving</span></p></li><ul style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Overspeeding by the Driver</span></p></li><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Drunk Driving</span></p></li><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Dangerous Driving : Wrong Lane Driving, Swirling, Unnecessary Overtaking</span></p></li></ul></ul><br /><ul style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:disc;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Unknown Route</span></p></li><li dir="ltr" style="list-style-type:disc;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Threatening / Indecent Behavior</span></p></li><ul style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Verbal</span></p></li><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Non Verbal</span></p></li><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Physical</span></p></li></ul></ul><ul style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:disc;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">External Threats</span></p></li><ul style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:circle;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Mob / Stalker</span></p></li></ul></ul><br /><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:italic;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Rash Driving</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Overspeeding</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"> can be curbed if we include a feature for Accelerometer which gives us real time speed and if a driver stays above the speed limit for more than 30 Seconds he gets notification that you are speeding. If he still doesn&rsquo;t reduce his speed for another 30 Seconds, a pre recorded call from &ldquo;Exotel&rdquo; will be triggered which again focuses on reducing the speed and also points out that non compliance will block his ID. In case the driver doesn&rsquo;t reduce, we call the customer to check if they are okay and if they would like to raise a formal complaint.</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">For </span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Drunk Driving</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"> or </span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Dangerous Driving</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">, customers can raise a complaint via SOS app post which the driver will be asked to follow the rules, a non compliant will end the ride there and a new driver can be assigned to the customer.&nbsp;</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:italic;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Unknown Route</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">If a driver takes a diversion/different route then the one chosen by the map and his current location is outside the deviation radius the customer should be notified, if he is okay, continue, else take driver side action message/call. (Explained in detail in feature list)</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:italic;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Threatening / Indecent Behavior</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Non Verbal </span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">actions include staring, indecent actions but not physically harming the customer which would usually make the customer uncomfortable. In such a case the customer will click on SOS and select the low level threat.</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Verbal</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"> actions include lewd language, arguments, indecent language, profanity etc. these incidents may turn into physical threatening too so we need to handle it properly therefore in such a case our functionality should allow the customer to record the conversation, the customer will click on SOS and select the medium level threat.</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Physical</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"> actions include showing weapons, manhandling, unwanted touches, indecent actions by driver, these will be under SOS severity level high and are the most critical to be dealt with plus escalated to proper authorities and supporting parties (police, namma warriors)</span></p><br /><br /><div dir="ltr" style="margin-left:0pt;" align="left">

SOS Threat Level | Actions
-- | --
Low | Get a Call Option (Prerecorded Exotel Call)Escalation to L0Call at end of ride to ask if safe
Medium | Get a Call Option (Prerecorded Exotel Call)Escalation to L1Call at end of ride to ask if safeShare ride details with Emergency Contacts
High | As soon as high chosen, live location and call to trusted contact Escalation to L2Get a Call Option (Prerecorded Exotel Call)Share ride detail, live location to nearby Police PatrolShare ride detail, live location to nearby Namma Warrior

<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-a470431b-7fff-1840-cf93-7700c670dddc"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:#ffff00;font-weight:700;font-style:italic;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Feature Details:</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:12pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Deviation Radius:</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">A circle of 500m Radius which allows drivers to take a shorter route to avoid traffic but not a different route itself. Once a driver is outside the radius for 15 seconds, the customer will get a notification asking if he is okay with the route. If the customer responds positively we continue to monitor, in case they say they are not okay, the driver should get a notification saying &ldquo;You are on a different route, go back to the original route&rdquo;. If within 1 minute the driver goes back to the circle, no action to be taken, otherwise an Exotel Call will be triggered on the same lines but adding that driver may be reprimanded for this behavior. In case he still doesn&rsquo;t comply a new driver will be assigned for the customer and the current ride would be ended after some distance near some public hotspot.</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Namma Warrior:</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Based on inputs from ARDU, our top 10 drivers, and ratings by women &amp; overall we decide Namma Warriors who are basically the caretakers of a specific area they usually operate out of. These will be the first responders from our side to high level SOS calls as they might know the area and can take action immediately.</span></p><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Get a Call Option</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Usually some indecent behaviors can be deterred if the driver feels the customer is talking with a friend/family member and may alert them if he takes any action. Thus, get a call option will be a pre-recorded call triggered from Exotel, it would be on the lines:</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">&ldquo;This is a pre-recorded message for your safety (to get them to be calm), please tell us your auto Number, and talk about what you are observing, please be on call till the ride ends / till when you feel safe&rdquo; this is to make sure the driver is deterred and customer can act normally.</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Exotel calls are also recorded so any arguments, comments etc by driver can also be recorded and will be stored in the DB</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Trusted Contacts</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">In high SOS situations, a customer may not get time to choose emergency contacts to whom to share ride details, scrolling from a list etc. Thus trusted contact is a single pre-feeded contact who will get the live location of ride, a call from exotel that you are a trusted contact of a rider and they have raised a SOS call to you, so that they get notified. They can then approach the police, call the customer and take appropriate action. This would reduce the critical time spent by a customer in choosing and help take actions faster.</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Police Partnerships:</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">As soon as a high SOS is chosen, through a communication channel we can notify the nearest police patrol station of driver, ride and vehicle details. This can be triggered if someone from L2 approves too.</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Late Night Confirmation:</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">For all rides after 10 PM to morning 7 AM can have a confirmation call with the customer asking if they reached home safely.</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:700;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Driver Trust Score:</span></p><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Driver will start with a 100 points Score, parameters affect the DTS in following way:</span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /><br /></span></p><div dir="ltr" style="margin-left:0pt;" align="left">

Parameter | Value Effect
-- | --
Rating | 5 and 4 have no effect, every 3 rating reduces it by -1, 2 rating by -2 and 1 rating by -4
Feedback | Sentimental Analysis to define whether a feedback by customer after ride to a driver was negative and deduct points accordingly
Cancellation Rate | After 75% Cancellation every 5% can deduct the DTS by 1 point
SOS raised | Every Low SOS deducts points by 3Every Medium SOS deducts points by 5Every High SOS deducts points by 10

</div><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"><br /></span><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">A DTS below 75 puts a driver on block for 10 days, post which he starts at 80 again.</span></p></b><br class="Apple-interchange-newline">

**Comment by VinayPratap97** [comment]
Adding my comments below :

--> It is observed that often times, there is a discrepancy in how drivers and customers share the incident. They deny the claims made by the other party. Hence, how will we solve for this ? ( Eg: Using past data and scores etc )
--> Defining speed limit and rash driving might be a bit tricky, accelerometer can be used for upper thresholds but speeding in crowded streets might still be unsafe. Once again we may need past data to make this more effective.
--> SOS with different levels of threat is not a good way if we are planning to have it in the customer UX. We can internally have it for our SOP and escalation matrix but its a negative experience to make a customer choose low threat when they are already in distress. It should happen in a more seamless way.
--> Namma Warriors are other autodrivers and are not lawkeepers. Hence involving them directly in a potential crime scene is a risky bet. They don't have any legal authority to handle any conflicts. Hence, it might be a better option to partner with private security agencies who have a legal capacity for conflict resolution or any such incidents
--> Get a call option is good as I have personally experienced this with many of my friends and family who want someone to stay on call till they reach the destination safely. 
--> The difference between trusted contacts and emergency contacts is not fully clear. Can be the same imo.
--> Late night call feature to confirm customer reached safely as done by rapido is good but it can be made more robust by including factors other than just night time, such as high risk areas, females or kids traveling alone, passengers with high belongings etc 
--> Need more clarity on the 100 point DTS score. Once a rating gets hit, there seems to be no way for the driver to increase the score again. Feedback based sentiment analysis can be thought through deeper to set some rules and objectivity. Not clear why cancellations is a part of this score.
--> Focus here is only on customer side. Similar options available for driver side as well ? particularly female drivers ?
--> Can look at female drivers to female passenger matching and other intelligent batching methods for high risk rides (eg. highly rated driver being mapped for female passenger traveling alone at night)

**Comment by VinayPratap97** [comment]
Also attaching the initial discussion points few months back to the same thread

Initial thoughts for solutioning: (Kindly read the below points with a pinch of salt as they are not well articulated solutions, rather points to give direction to think and debate which will later translate into product requirements)

→ We can classify certain trips as higher risk compared to others : 
Females traveling late at night
Passengers having high value belongings 
Children traveling by themselves
Pickup/drop in potential risk areas
For these trips, there needs to be a separate protocol. Eg:, 
We may ask for selfie, photo or some other identity verification for trip start
Camera access to record at any emergency instance during the trip
Separate dashboard to monitor such trips and raise alerts based on some preset parameters such as time or distance exceeding the estimate
Sharing trip status with emergency contacts


→ Screening process for drivers: This is a very important step in the prevention process for which we are not having any measures currently. Criminal and background check is a cost-prohibitive feature but cannot be compromised completely. Let’s explore solutions for this process.


→ Screening process for consumers: Drivers' interest also has to be protected. Maybe this can be introduced only for customers who get low ratings consistently or who have a pickup or drop in a location identified as high risk. 

→ Increasing predictability, Raising alerts: Since the focus is on prevention, we need to be able to anticipate potential safety threats or incidents. Live Dashboard for high deviation/delay rides is one such measure. SOS button is another such feature. We can think of more options here.

→ Using data to keep check: We also need to monitor the physical and mental state of our drivers who are providing the service, Fatigue management to prevent accidents. eg have limit on number of trips, distance a driver can drive in a day

→ Safety guidelines: Educating the customers of the available safety measures is very important. Eg. 
Check photo/id card of driver before trip start 
Don’t share personal details
Don’t leave any belongings, check well after trip

→ Effectively using the rating mechanism: We need to build a robust rating system for both drivers and passengers. It will help us identify bad behavior based on which we can implement control mechanisms. 

→ Encouraging service providers to have physical safety tools within their vehicles. Eg: Pepper Spray. Need to consider how auto drivers will perceive this and passengers should not misuse it. 

→ While our solutioning needs to prevent any negative incidents, we also need to have a protocol for when an incident has happened or happening. 


→ Intelligent driver pooling for high-risk rides

**Reply by VinayPratap97** [reply]
Customer SOS Functionally:

On click of the SOS button
Trigger SMS to all their emergency contacts
Trigger FCM to all their emergency contacts
Trigger an email to the designated Juspay employees as a SEV1 which will eventually trigger a call to them as well
As usual the call will also be triggered to the control room

SOS Feature Availability

SOS button will be available once the OTP is verified and the ride has started. The SOS button will be available only while the trip is in progress.

SMS or FCM or Email Content:Nc
Emergency: Namma Yatri Customer is currently in a SOS situation. Please call the customer: <number>
You can see trip details here: <app_link>

Additionally in email:
Please add 3 sections namely:

Customer info: 
Name
Mobile
Email

Driver info:
Name
Mobile
Email

Vehicle number
DL number
Activated on
Number of trips

Trip info:
From
To
Current location
Route map

**Reply by VinayPratap97** [reply]
Ola Implementation

https://blog.olacabs.com/how-to-use-the-sos-button-on-the-ola-app/

Operations Protocol To be Followed

The control room staff should look for the SOS trip details on email
As a first step, the control room staff should reach the customer phone and assess what help is needed
If the customer is not reachable, the control room staff should reach the driver’s mobile phone and ask for the call to be passed to the rider
The control room staff should dial for police assistance if any of the following events occur
Unable to reach customer directly or through driver
Unable to reach either driver or rider
Rider explicitly states that their safety is in doubt

How to reach police control room/patrol service
